static/js/tools/choropleth/choropleth.tsx

```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
     this.mouseOutAction(index);
   };
 
+  /**
+   * Redirects the webclient to a particular geo. Handles breadcrumbs in redirect.
+   * @param {string} geoDcid to redirect to.
+   * @param {string} curGeo human-readable current geo, e.g. United States when geoDcid
+   *                 is country/USA.
+   */
+  private redirectToGeo = (geoDcid: string, curGeo: string): void => {
+    const url = new URL(window.location.href);
+    const currGeoId = url.searchParams.get("geoDcid");
+    let baseUrl = "/tools/choropleth";
+    baseUrl += buildChoroplethParams(["statVar", "pc", "mdom"]);
+    baseUrl += "&geoDcid=" + geoDcid;
+    baseUrl += "&bc=";
+
+    const idxOfCurr = this.state.breadCrumbs.findIndex(
+      (crumb) => crumb.geoId === geoDcid
+    );
+    let breadCrumbsList = this.state.breadCrumbs;
+    if (idxOfCurr == -1) {
+      breadCrumbsList.push({ geoId: currGeoId, geoName: curGeo });
+    } else {
+      breadCrumbsList = breadCrumbsList.slice(0, idxOfCurr);
+    }
+    this.setState({
+      breadCrumbs: breadCrumbsList,
+    });
+    for (let i = 0; i < breadCrumbsList.length; i++) {
+      baseUrl += i === 0 ? "" : ";";
+      baseUrl += breadCrumbsList[i].geoId + "~" + breadCrumbsList[i].geoName;
+    }
+    history.pushState({}, null, baseUrl);
+    this.loadGeoJson();
+  };
+
+  /**
+   * Generates the breadcrumbs text from browser url.
+   * @param {string} curGeo human-readable current geo to display
+   * at end of list of hierarchy of locations.
+   */
+  private generateBreadCrumbs = (curGeo: string): void => {
+    const url = new URL(window.location.href);
+    const breadcrumbs = url.searchParams.get("bc");
+    const breadcrumbsDisplay = [];
+    if (breadcrumbs) {
+      const crumbs = breadcrumbs.split(";");
+      crumbs.forEach((crumb) => {
+        // The geoDcid reference and human-readable curGeo are separated by a '~'.
+        const [levelRef, humanName] = crumb.split("~");
+        if (levelRef) {
+          breadcrumbsDisplay.push(
+            <span
+              key={levelRef}
+              className="clickable-link"
+              onClick={() => this.redirectToGeo(levelRef, curGeo)}
+            >
+              {humanName + " > "}
+            </span>
+          );
+        }
+      });
+      breadcrumbsDisplay.push(<span key={curGeo}>{curGeo}</span>);
+    }
+    // Add breadcrumbs + current geoId
+    // Example: "USA" > "FL" > "Miami-Dade"
+    // Where Miami-Dade is the current id
+    ReactDOM.render(breadcrumbsDisplay, document.getElementById("breadcrumbs"));
```

Comment:
Might not be good to call ReactDOM.render in a react Component. How about set the state of a children component for the bradcrumbs?